src/main/java/org/fungover/thunder/Topic.java

```diff
@@ -0,0 +1,76 @@
+package org.fungover.thunder;
+
+public record Topic(String name) {
+
+
+    public static Topic create(String topicName) {
+        checkTopicNamingValidation(topicName);
+        return new Topic(topicName);
+    }
+
+    private static void checkTopicNamingValidation(String topicName) {
+        if (topicName == null || topicName.isEmpty())
+            throw new IllegalArgumentException("Invalid topic: Should contain at least one character");
+
+        if (topicName.startsWith("/"))
+            throw new IllegalArgumentException("Invalid topic: Can not begin with '/'");
+
+        if (topicName.contains("$"))
+            throw new IllegalArgumentException("Invalid topic: Can not contain '$'");
+
+        if (!isAValidTopicName(topicName))
+            throw new IllegalArgumentException("Invalid topic: Topic name does not follow MQTT topic naming conventions");
+    }
+
+    //checks a created topic name if its valid
+    public static boolean isAValidTopicName(String topicName) {
+        long count = 0L;
+        int numInPart = 0;
+
+        if (topicName.contains("#")) {
+            count = topicName
+                .chars()
+                .filter(c -> c == '#')
+                .count();
+        }
```

Comment:
I believe the 'if' statement is unnecessary  as the 'filter' function already excludes any topics containing the '#' symbol ?

The test also passes without the need for an 'if' statement" so don't know if it have any impact or not... But good work anyway
```suggestion
           long  count = topicName
                .chars()
                .filter(c -> c == '#')
                .count();
      
```

Reply:
We implemented it cause it was unnecessary for the string to get every char checked for multi level wildcards if it didn't contain any from start 😊 and thanks again for the code!